examples/Example14/CommonStruct.h

```diff
@@ -29,12 +29,34 @@ struct TrackData {
       : position{x, y, z}, direction{dirx, diry, dirz}, energy{ene}, pdg{pdg_id}
   {
   }
+
+  inline bool operator<(TrackData const &t)
+  {
+    if (pdg < t.pdg) return true;
+    if (pdg > t.pdg) return false;
+    if (energy < t.energy) return true;
+    if (energy > t.energy) return false;
+    if (position[0] < t.position[0]) return true;
+    if (position[0] > t.position[0]) return false;
+    if (position[1] < t.position[1]) return true;
+    if (position[1] > t.position[1]) return false;
+    if (position[2] < t.position[2]) return true;
+    if (position[2] > t.position[2]) return false;
+    if (direction[0] < t.direction[0]) return true;
+    if (direction[0] > t.direction[0]) return false;
+    if (direction[1] < t.direction[1]) return true;
+    if (direction[1] > t.direction[1]) return false;
+    if (direction[2] < t.direction[2]) return true;
+    if (direction[2] > t.direction[2]) return false;
+    return false;
+  }
```

Comment:
That's a bit un-idiomatic to write the `operator<` (I think I've seen `if (pdg != t.pdg) return pdg < t.pdg;` more often), but ok for now.

Reply:
Indeed nicer this way, adopted it